settings.gradle

```diff
@@ -1,22 +1,17 @@
 import com.gradle.scan.plugin.PublishedBuildScan
 
-
 pluginManagement {
     repositories {
+        // # Gradle looks for dependency artifacts in repositories listed in 'repositories' blocks in descending order.
+
+        // ## Prefer repos controlled by Airbyte.
+        // TODO: add airbyte-controlled proxy repos here
```

Comment:
❓How come we have this comment so many times?

Reply:
Unless I missed something, you can't actually define all repositories in just one place in gradle. You can for regular dependencies, but then the build files require another, and the custom plugins in `buildSrc` yet another. It's really dumb IMHO.

The hierarchy of the repos is the same in all cases: first look locally, then look into our cloudrepo server, then into public repos in decreasing order of likelihood. When resolving a dependency, gradle iterates over each repo that's not filtered out by the `content { ... }` block and performs an HTTP GET to that server. This is predictably fast when the server is Maven Central but can be really slow when it's some random corporate server which could be anywhere on the planet running on some vm in a box on top of a dusty filing cabinet.

I could add this as commentary if it helps! LMK

Reply:
Copy and paste sounds right here. No need to expand :)